File: example/src/test/java/com/example/util/excel/TestExportMap.java
package com.example.util.excel;

import com.zipe.util.doc.ExcelUtil;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.util.ArrayList;
import java.util.Date;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

class TestExportMap {

  @Test
  void exportXls() throws IOException {
    List<Map<String, Object>> list = new ArrayList<>();
    Map<String, Object> map = new LinkedHashMap<>();
    map.put("name", "");
    map.put("age", "");
    map.put("birthday", "");
    map.put("sex", "");
    Map<String, Object> map2 = new LinkedHashMap<String, Object>();
    map2.put("name", "測試是否是中文長度不能自動寬度.總統府地址:臺北市中正區重慶南路一段122號。");
    map2.put("age", null);
    map2.put("sex", null);
    map.put("birthday", null);
    Map<String, Object> map3 = new LinkedHashMap<String, Object>();
    map3.put("name", "張三");
    map3.put("age", 12);
    map3.put("sex", "男");
    map3.put("birthday", new Date());
    list.add(map);
    list.add(map2);
    list.add(map3);
    Map<String, String> map1 = new LinkedHashMap<>();
    map1.put("name", "姓名");
    map1.put("age", "年齡");
    map1.put("birthday", "出生日期");
    map1.put("sex", "性别");
    File f = new File("D:/test.xls");
    OutputStream out = new FileOutputStream(f);
    ExcelUtil.exportExcel(map1, list, out);
    out.close();
  }
}

File: example-kotlin/src/test/kotlin/com/example/util/excel/TestExportMap.kt
package com.example.util.excel

import com.zipe.util.doc.ExcelUtil
import io.kotest.core.spec.style.FunSpec
import java.io.File
import java.io.FileOutputStream
import java.util.Date

/**
 * 以 List&lt;Map&gt; 為資料來源的 Excel 匯出測試，標題列同樣以排序 Map 指定。
 */
class TestExportMap : FunSpec({

    test("exportXls") {
        val list = mutableListOf<Map<String, Any?>>()
        val map = linkedMapOf<String, Any?>(
            "name" to "",
            "age" to "",
            "birthday" to "",
            "sex" to "",
        )
        val map2 = linkedMapOf<String, Any?>(
            "name" to "測試是否是中文長度不能自動寬度.總統府地址:臺北市中正區重慶南路一段122號。",
            "age" to null,
            "sex" to null,
        )
        map["birthday"] = null
        val map3 = linkedMapOf<String, Any?>(
            "name" to "張三",
            "age" to 12,
            "sex" to "男",
            "birthday" to Date(),
        )
        list.add(map)
        list.add(map2)
        list.add(map3)
        val map1 = linkedMapOf(
            "name" to "姓名",
            "age" to "年齡",
            "birthday" to "出生日期",
            "sex" to "性别",
        )
        val f = File("D:/test.xls")
        FileOutputStream(f).use { out ->
            ExcelUtil.exportExcel(map1, list, out)
        }
    }
})
